packages/rum-core/src/domain/action/trackClickActions.ts

```diff
@@ -209,21 +213,33 @@ function startClickAction(
 
 type ClickActionBase = Pick<ClickAction, 'type' | 'name' | 'target' | 'position'>
 
-function computeClickActionBase(event: MouseEventOnElement, actionNameAttribute?: string): ClickActionBase {
+function computeClickActionBase(event: MouseEventOnElement, configuration: RumConfiguration): ClickActionBase {
   const rect = event.target.getBoundingClientRect()
+  const privacyEnabledForActionName =
+    (getNodePrivacyLevel(event.target, configuration.defaultPrivacyLevel) === NodePrivacyLevel.HIDDEN ||
+      configuration.defaultPrivacyLevel === DefaultPrivacyLevel.MASK) &&
+    configuration.enablePrivacyForActionName
+  const actionName = getActionNameFromElement(
+    event.target,
+    configuration.actionNameAttribute,
+    privacyEnabledForActionName
+  )
+  const target = {
+    width: Math.round(rect.width),
+    height: Math.round(rect.height),
+    selector: getSelectorFromElement(event.target, configuration.actionNameAttribute),
+  }
+
   return {
     type: ActionType.CLICK,
-    target: {
-      width: Math.round(rect.width),
-      height: Math.round(rect.height),
-      selector: getSelectorFromElement(event.target, actionNameAttribute),
-    },
+    target:
+      actionName === ACTION_NAME_PLACEHOLDER && privacyEnabledForActionName ? assign({ masked: true }, target) : target,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏This check `actionName === ACTION_NAME_PLACEHOLDER` makes me wondering if the privacy logic could not be collocated to the action name computation. Something like
```ts
function getActionNameFromElement(element: Element, configuration: RumConfiguration): { value: string, masked: boolean }
```